GeometricSeqPhenotype.java

```diff
@@ -0,0 +1,308 @@
+    // for all indexSite such that gsp.sequence.charAt(i):
+    //     indexSite is a character in ALPHABET
+    // for all codon such that gsp.sequence.charAt(i) + gsp.sequence.charAt(i + 1) + gsp.sequence.charAt(i + 2):
+    //     codon is not "STOP"
+
+    /**
+     * Constructor that creates a new GeometricSeqPhenotype.
+     *
+     * @effects Constructs a new GeometricSeqPhenotype that has a random starting sequence of length Parameters.sequence.length()
+     * and an empty GeometricPhenotype
+     */
+    public GeometricSeqPhenotype() {
+        this.nucleotideSequence = new char[Parameters.startingSequence.length()];
+        startingSequenceGenerator();
+        checkRep();
+    }
+
+    /**
+     * Constructor that creates a new GeometricSeqPhenotype.
+     *
+     * @param tA
+     * @param tB
+     * @effects Constructs a new GeometricSeqPhenotype that has a random starting sequence of length Parameters.sequence.length()
+     * and GeometricPhenotype with the data content of the given parameters.
+     */
+    public GeometricSeqPhenotype(double tA, double tB) {
+        traitA = tA;
+        traitB = tB;
+
+        startingSequenceGenerator();
+        checkRep();
+    }
+
+    // Generates a random sequence of nucleotides of length Parameters.startingSequence.length()
+    private void startingSequenceGenerator() {
+        char[] nucleotideAlphabet = Parameters.AlphabetType.NUCLEOTIDES.getValidCharacters().toCharArray();
+
+        for (int siteIndex = 0; siteIndex < Parameters.startingSequence.length(); siteIndex += 3) {
+            String aminoAcid = "";
+
+            // Generate codon at siteIndex
+            while (aminoAcid.equals("") || aminoAcid.equals("STOP")) {
+                String codon = "";
+                for (int codonIndex = 0; codonIndex < 3; codonIndex++) {
+                    int indexAlphabet = random(nucleotideAlphabet.length);
+                    char nucleotide = nucleotideAlphabet[indexAlphabet];
+                    codon += nucleotide;
+                    this.nucleotideSequence[siteIndex + codonIndex] = nucleotide;
+                }
+                aminoAcid = Biology.CodonMap.CODONS.getAminoAcid(codon);
+            }
+        }
+    }
+
+    /**
+     * Constructor that creates a new GeometricSeqPhenotype.
+     *
+     * @param tA
+     * @param tB
+     * @param startingSequence
+     * @requires nucleotideSequence != null && nucleotideSequence.length() > 0 && nucleotideSequence.length() % 3 == 0
+     * @effects Constructs a new GeometricSeqPhenotype that has a starting sequence
+     * and GeometricPhenotype with the data content of the given parameters.
+     */
+    public GeometricSeqPhenotype(double tA, double tB, char[] startingSequence) {
+        traitA = tA;
+        traitB = tB;
+        this.nucleotideSequence = startingSequence;
+        checkRep();
+    }
+
+    public GeometricSeqPhenotype(double tA, double tB, char[] startingSequence, int e, int ne) {
```

Comment:
I hate to be tedious here, but could we add docstrings to this constructor as well?

Reply:
oops--nevermind. Looks like Hugh already asked